src/tracing/src/lib.rs

```diff
@@ -57,27 +84,67 @@ impl RwTracingService {
         }
     }
 
+    #[cfg(madsim)]
+    fn start_tracing_listener(
+        _rx: UnboundedReceiver<Collector>,
+        _print_to_console: bool,
+        _slow_request_threshold_ms: u64,
+        _jaeger_addr: Option<SocketAddr>,
+    ) -> JoinHandle<()> {
+        unreachable!()
+    }
+
+    #[cfg(not(madsim))]
     fn start_tracing_listener(
         rx: UnboundedReceiver<Collector>,
+        print_to_console: bool,
         slow_request_threshold_ms: u64,
+        jaeger_addr: Option<SocketAddr>,
     ) -> JoinHandle<()> {
+        use futures::StreamExt;
+        use rand::Rng;
+
         tracing::info!(
-            "tracing service started with slow_request_threshold_ms={slow_request_threshold_ms}"
+            "tracing service started with slow_request_threshold_ms={slow_request_threshold_ms}, print_to_console={print_to_console}"
         );
+
         std::thread::Builder::new()
             .name("minitrace_listener".to_string())
             .spawn(move || {
                 let func = move || {
-                    let stream = rx.for_each_concurrent(None, |collector| async move {
+                    let rt = tokio::runtime::Builder::new_current_thread().build()?;
+                    let stream = rx.for_each_concurrent(None, |collector| async {
                         let spans = collector.collect().await;
-                        if let Some(span) = spans.first() {
-                            // print requests > 100ms
-                            if span.duration_ns >= slow_request_threshold_ms * 1_000_000 {
-                                tracing::info!("{:?}", spans);
+                        println!("{:?}", spans);
```

Comment:
Use tracing::debug?

Reply:
I expect this to be effective in release mode

Reply:
Yeah, but we should use tracing::info for sure